02_classify_stratified/nbconverted/plot_purity_results.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Plot tumor purity prediction results
+# 
+# Here, we'll visualize the results of our tumor purity prediction. For now, we're trying to predict whether a given sample is above or below the median tumor purity (i.e. classification).
+# 
+# Results analyzed here are generated by the `run_purity_prediction.py` script.
+
+# In[1]:
+
+
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+import mpmp.config as cfg
+import mpmp.utilities.analysis_utilities as au
+
+
+# In[2]:
+
+
+# set results directory
+results_dir = Path(cfg.results_dir,
+                   'purity_binarized_preds',
+                   'purity').resolve()
+
+
+# In[3]:
+
+
+results_df = (
+    au.load_purity_binarized_results(results_dir)
+      .drop(columns=['identifier'])
+)
+# all me_450k results are compressed
+results_df[results_df.training_data == 'me_450k'] = 'me_450k_compressed'
+print(results_df.shape)
+print(results_df.training_data.unique())
+results_df.head()
+
+
+# In[4]:
+
+
+sns.set({'figure.figsize': (8, 6)})
+sns.set_style('whitegrid')
+plot_df = (
+    results_df[results_df.data_type == 'test']
+      .sort_values(by=['signal', 'training_data'])
+)
+
+sns.boxplot(data=plot_df, x='signal', y='aupr', hue='training_data')
+plt.title('Binarized tumor purity prediction performance, by data type')
+plt.legend(title='Data type')
+plt.xlabel('Signal or shuffled')
+plt.ylabel('AUPR')
+
+
+# In[5]:
+
+
+# now facet by cancer type
```

Comment:
I know nothing about cancers. But I wonder why the performance is so low with all data types for PAAD.

Reply:
I don't know much about pancreatic cancer specifically (and it's debatable whether I know much more than nothing about cancer in general, haha) - but looking at [this page on the TCGA pancreatic cancer study](https://www.cancer.gov/about-nci/organization/ccg/research/structural-genomics/tcga/studied-cancers/pancreatic), it seems like pancreatic tumors tend to have very low purity in general (or in other words, PAAD samples are generally heterogeneous mixtures of stromal cells and cancerous cells, rather than samples of mostly cancerous cells like in many other cancer types).

It's not exactly intuitive to me why this would lead to poor performance: ideally our model would pick up on this and predict lower values for PAAD samples, but maybe the high overall tumor heterogeneity makes it hard to distinguish between samples based just on -omics data? I dunno, I'll have to think it through.

Reply:
Ah, super interesting! I wonder if using other binarization strategies would improve results with PAAD (not sure why it would, but just wondering).

Reply:
Yeah, this would be interesting to look at! We'll see if regression improves things, but other binarization strategies would be interesting too.